### PR TITLE
Draft M3GNet Implementation

### PR DESCRIPTION
**Pull request by stefanbringuier**
### Overview

Thanks @jla-gardner  for a great package. This aim of this PR is to introduce the **M3GNet** architecture into `graph-pes` as an attempt to  replicate the original design from [Chen & Ong, Nature Computational Science (2022)](https://doi.org/10.1038/s43588-022-00349-3). 

### Key Changes
- **New Module:** [src/graph_pes/models/m3gnet.py](https://github.com/stefanbringuier/graph-pes/blob/m3gnet_model/src/graph_pes/models/m3gnet.py)
- **M3GNetInteraction Class:** Implements three-body message passing with radial basis functions.
- **M3GNet Model:** Stacks interaction layers with chemical embedding, energy scaling, and a readout MLP.

### Review Assistance Required
- [ ]  Validate **accuracy** against the original [M3GNet](https://github.com/materialsvirtuallab/m3gnet) design.
- [ ]  Ensure **correctness** of three-body interactions, radial basis functions, and overall architecture.
- [ ]  Solicit **feedback** on implementation design and quality.

### Testing Status
-  [x] Simple shape testing of energy, forces ( [src/tests/test_m3gnet.py](https://github.com/stefanbringuier/graph-pes/blob/m3gnet_model/tests/models/test_m3gnet.py))
-  [ ] Compare model predictions.

### Next Steps
1. Incorporate review feedback.
2. Benchmark against reference implementations (require training?)

> [!NOTE]
> This PR is really for **review assistance and refinement**, not immediate merging.

### CODE REVIEW

**Comment by jla-gardner** [comment]
Hi Stefan 👋

Thank you for the contribution! I'll need to take some time here to understand the original m3gnet implementation, and then to compare it to yours.

Style-wise, your implementation looks great! To aid with others reading this in the future, could I please ask that you leave occasional comments with the expected shape of various tensors within the forward passes of your components. 

I see that you have also made use of a Cosine smoothing function: feel free to add that as a class in `models/components/distances.py` if you wanted.

Before I train on some test systems, could I also ask you to do some parameter counting: using the same hyperparameter settings, do you get (roughly) the same number of parameters in the `graph-pes` and `M3GNet` implementations?

Thank you again!

**Comment by stefanbringuier** [comment]
> ... could I please ask that you leave occasional comments with the expected shape of various tensors within the forward passes of your components.

Yes will do this will help me remember as well.

> Before I train on some test systems, could I also ask you to do some parameter counting: using the same hyperparameter settings, do you get (roughly) the same number of parameters in the `graph-pes` and `M3GNet` implementations?

Yes this needs to be done.

**Comment by stefanbringuier** [comment]
Added comments in `m3gnet.py` on expected shapes.

When I reviewed the checkpoint model [`M3GNet-MP-2021.2.8-PES`](https://github.com/materialsvirtuallab/matgl/tree/main/pretrained_models/M3GNet-MP-2021.2.8-PES) the expected number of trainable parameters looks to be `288,157`. The current implementation using the same number of channels, layers, etc is only giving `51,509`. When I further dug down into the model structure, I think my implementation is wrong w.r.t:

- Not using GatedMLPs
- Readout MLP is shallower
- Three body message construction not exactly the same

Will need some time to debug through it.
